test/functional/tests/io/trim/test_trim.py

```diff
@@ -0,0 +1,252 @@
+                                                     core_dev.size * 0.9)
+        occupancy_before = core.get_occupancy()
+        TestRun.LOGGER.info(str(core.get_statistics()))
+
+    with TestRun.step("Start blktrace monitoring on all devices."):
+        blktraces = start_monitoring(core_dev, cache_dev, core)
+
+    with TestRun.step("Remove file."):
+        os_utils.sync()
+        os_utils.drop_caches()
+        test_file.remove()
+
+    with TestRun.step(
+            "Ensure that discards were detected by blktrace on proper devices."):
+        discard_expected = {"core": discard_support["core"], "cache": False, "cas": True}
+        stop_monitoring_and_check_discards(blktraces, discard_expected)
+
+    with TestRun.step("Ensure occupancy reduced."):
+        occupancy_after = core.get_occupancy()
+        TestRun.LOGGER.info(str(core.get_statistics()))
+
+        if occupancy_after >= occupancy_before:
+            TestRun.LOGGER.error("Occupancy on core after removing test file greater than before.")
+        else:
+            TestRun.LOGGER.info("Occupancy on core after removing test file smaller than before.")
+
+    with TestRun.step("Check CAS sysfs properties values."):
+        check_sysfs_properties(cache, cache_dev, core, core_dev.parent_device,
+                               core_supporting_discards=discard_support["core"])
+
+
+def check_sysfs_properties(cache, cache_dev, core, core_disk, core_supporting_discards):
+    expected_discard_max_bytes = int(core_disk.get_discard_max_bytes()) \
+        if core_supporting_discards else int(cache_dev.size.get_value())
+    cas_discard_max_bytes = int(core.get_discard_max_bytes())
+    compare_properties(cas_discard_max_bytes, expected_discard_max_bytes, "discard_max_bytes")
+
+    expected_discard_granularity = int(core_disk.get_discard_granularity()) \
+        if core_supporting_discards else int(cache.get_cache_line_size())
+    cas_discard_granularity = int(core.get_discard_granularity())
+    compare_properties(
+        cas_discard_granularity, expected_discard_granularity, "discard_granularity")
+
+    cas_discard_zeroes_data = int(core.get_discard_zeroes_data())
+    if cas_discard_zeroes_data == 0:
+        TestRun.LOGGER.info("CAS discard_zeroes_data value equals 0 as expected.")
+    else:
+        TestRun.LOGGER.error(f"CAS discard_zeroes_data value equals {cas_discard_zeroes_data}. "
+                             "Expected value for this property is 0.")
+
+
+def compare_properties(value, expected_value, property_name):
+    if expected_value == value:
+        TestRun.LOGGER.info(f"CAS {property_name} value is correct.")
+        return
+    TestRun.LOGGER.error(f"CAS property {property_name} value equals {value} and differs "
+                         f"from expected value: {expected_value}.")
+
+
+def stop_monitoring_and_check_discards(blktraces, discard_support):
+    time.sleep(10)
+    os_utils.sync()
+    os_utils.drop_caches()
```

Comment:
Please, add 1-2 seconds of sleep after `drop_caches()` - some DUTs, especially with RAIDs, need it.